solutions/usaco-832.mdx

```diff
@@ -22,6 +22,106 @@ In this problem, there are three cases that we have to consider:
 
 <LanguageSection>
 
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int main() {
+	freopen("milkorder.in", "r", stdin);
+	freopen("milkorder.out", "w", stdout);
+
+	int n, m, k;
+	cin >> n >> m >> k;
+
+	vector<int> pos(n + 1, -1);
+	map<int, int> statuspos;
+	vector<int> status(m);
+	for (int i = 0; i < m; i++) {
+		cin >> status[i];
+		statuspos[status[i]] = i;
+	}
+
+	set<int> exist;
+	for (int i = 0; i < k; i++) {
+		int c, p;
+		cin >> c >> p;
+		pos[p] = c;
+		exist.insert(c);
+	}
+	
+	// Case 1
+	for (int i = 1; i <= n; i++) {
+		if (pos[i] == 1) {
+			cout << i << endl;
+			return 0;
+		}
+	}
+	
+	// Case 2
+	if (statuspos.count(1)) {
+		int start = statuspos[1] - 1;
+		while (start >= 0) {
+			if (exist.count(status[start])) {
+				break;
+			}
+			start--;
+		}
+		bool check = start == -1;
```

Comment:
```suggestion
		bool check = (start == -1);
```

Reply:
also use a better variable name than `check`

Reply:
parenthesis doesn't really matter

Reply:
makes it clearer since there are a lot of equal signs

Reply:
ykw personal preference, op i'll leave it up to you